File: CS2613/labs/L18/fib.py
class Fib:
	def __init__(self,max):
        	self.max = max
        	self.a = 0
        	self.b = 1

	def __next__(self):
		if self.a < self.max:
			tmp = self.a
			self.a = self.b
			self.b += tmp
			#self.a,self.b = self.b, self.a+self.b
			return tmp
		else:
			raise StopIteration

	def __iter__(self):
		self.a = 0
		self.b = 1
		return self

File: CS2613/labs/L18/fib.m
function ret = fib(n)
  a = 0;
  b = 1;
  for i=0:n
     tmp = a;
    a = b;
    b = b +tmp;
  endfor
  ret = tmp;
  
endfunction

%!assert (fib(0) == 0);
%!assert (fib(1) == 1);
%!assert (fib(2) == 1);
%!assert (fib(3) == 2);
%!assert (fib(4) == 3);
%!assert (fib(5) == 5);